assets/components/src/plugin-settings/index.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * WordPress dependencies.
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { Component, Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies.
+ */
+import { SectionHeader, Notice } from '../';
+import SettingsSection from './SettingsSection';
+
+class PluginSettings extends Component {
```

Comment:
Why should we avoid class components?

Reply:
Good question! Mainly for consistency, simplicity, and readability; but also React as a project is embracing a more functional style (thinking here about hooks, mainly). I can't think of any reason to use class-based components: it's more code to write (and read) and more of a cognitive load to follow what happens in the code – but this might be personal preference.

Reply:
Makes sense. Although I like the functional style, I still think class components improve readability and enforce good practices when a component contains business logic, which is usually called container components.

This is likely personal preference but also changes according to the rest of the stack. A redux-based application probably shouldn't have business logic inside react components and should favor function components to encourage simplification and the overall presentational aspect.